client/lib/purchases/index.ts

```diff
@@ -100,32 +114,34 @@ function getSubscriptionsBySite( subscriptions ) {
 		.sort( ( a, b ) => ( a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1 ) );
 }
 
-function getName( purchase ) {
+export function getName( purchase: Purchase ): string {
 	if ( isDomainRegistration( purchase ) || isDomainMapping( purchase ) ) {
-		return purchase.meta;
+		return purchase.meta ?? '';
 	}
 	return purchase.productName;
 }
 
-function getDisplayName( purchase ) {
+export function getDisplayName( purchase: Purchase ): ReactChild | string | undefined {
 	const jetpackProductsDisplayNames = getJetpackProductsDisplayNames();
 	if ( jetpackProductsDisplayNames[ purchase.productSlug ] ) {
 		return jetpackProductsDisplayNames[ purchase.productSlug ];
 	}
 	return getName( purchase );
 }
 
-function getPartnerName( purchase ) {
+export function getPartnerName( purchase: Purchase ): string | null {
 	if ( isPartnerPurchase( purchase ) ) {
 		return purchase.partnerName;
 	}
 	return null;
 }
 
 // TODO: refactor to avoid returning a localized string.
-function getSubscriptionEndDate( purchase ) {
+export function getSubscriptionEndDate( purchase: Purchase ): string {
 	const localeSlug = i18n.getLocaleSlug();
-	return moment( purchase.expiryDate ).locale( localeSlug ).format( 'LL' );
+	return moment( purchase.expiryDate )
+		.locale( localeSlug ?? 'en' )
```

Comment:
`i18n.getLocaleSlug()` can apparently return `undefined`, and `moment.locale()` requires a string, so I set a default here.